src/compile/legend/parse.ts

```diff
@@ -176,5 +176,21 @@ export function mergeLegendComponent(mergedLegend: LegendComponent, childLegend:
     );
     mergedLegend.setWithExplicit(prop, mergedValueWithExplicit);
   }
+
+  if (mergedLegend.implicit.encode.symbols.update.opacity && mergedLegend.implicit.opacity) {
```

Comment:
Don't add this HACK here.  

Add the rule inside the `legend/encode.ts` for the symbols method. (If there is no `symbols` method, create one so we have a consistent pattern to determine default values of things in encode block.)